### DISCUSSION TITLE
Error on redirects file

### DISCUSSION THREAD

**Discussion by RigottiG** [vercel/next.js#39762] [Help] [answered]
I need to do a redirect in my application, it has the following structure.
`/search/shoes?variation=111`

But it is returning the following error
> Reason: Unexpected MODIFIER at 85, expected END
`/search/shoes**?**variation=111` 

How could I fix ? The error appears to be due to the  **"?"** 

```
{
 source:  `/search/shoes?variation=111`,
 destination: `/category/shoes`,
 permanent: true   
}
```

**Comment by gaston-flores** [comment, marked as answer]
You can't add the query to `source` directly, [you need to use `has`](https://nextjs.org/docs/api-reference/next.config.js/redirects#header-cookie-and-query-matching). In your case it would be
```js
module.exports = {
  async redirects() {
    return [
      {
        source: '/search/shoes',
        has: [
          {
            type: 'query',
            key: 'variation',
            value: '111',
          },
        ],
        destination: '/category/shoes',
        permanent: true,
      },
    ]
  },
}
```
